### PR TITLE
Unbinding subject from thread context at beginning of filter chain.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

This change is explicitly unbinding the subject at the _beginning_ of the filter chain. Previously this happened through a `ContainerResponseFilter`, which is not always called.

/nocl

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/shared/security/ShiroSecurityBinding.java [hunk 1]
```diff
@@ -59,8 +59,5 @@ public void configure(ResourceInfo resourceInfo, FeatureContext context) {
             LOG.debug("Resource method {}#{} requires an authorization checks.", resourceClass.getCanonicalName(), resourceMethod.getName());
             context.register(new ShiroAuthorizationFilter(requiresPermissions));
         }
-
-        // TODO this is the wrong approach, we should have an Environment and proper request wrapping
-        context.register((ContainerResponseFilter) (requestContext, responseContext) -> ThreadContext.unbindSubject());
```
Comment (mpfz0r) [inline graylog2-server/src/main/java/org/graylog2/shared/security/ShiroSecurityBinding.java:64]: hmm, i guess it can't hurt to keep this as well, no?

Comment (dennisoelkers) [reply graylog2-server/src/main/java/org/graylog2/shared/security/ShiroSecurityBinding.java:64]: I do not think it makes sense to keep it when there are requests where it is just not executed, in theses cases it provides a false sense of security.

Comment (mpfz0r) [reply graylog2-server/src/main/java/org/graylog2/shared/security/ShiroSecurityBinding.java:64]: fair enough

**Review by mpfz0r — APPROVED** [review]
LGTM. I think we should try to do this like suggested here, if possible.
https://shiro.apache.org/subject.html#Subject-ThreadAssociation

**Pull request merged**